network.cpp

```diff
@@ -189,6 +224,44 @@ bool net_behaves_like_the_function(const network &n, const std::function<double(
 
 }
 
+bool all_weigths_are_active(const network &n)
+{
+  std::vector<std::vector<std::vector<weight>>> weights = n.get_net_weights();
+
+  for(auto &layer : weights ){
+      for(auto &node : layer){
+          for (auto &weight : node){
+              if(!weight.is_active()){
+                  return false;
+                }
+            }
+        }
+    }
+  return true;
+}
+
+bool on_average_an_nth_of_the_activations_are_mutated(const network &n, const std::vector<weight>&registered_mutations,
+                                                      const double &mut_rate, int repeats)
+{
+  int number_of_weights = get_number_weights(n);
+  size_t number_mutations = registered_mutations.size();
+  double error = 0.1 * repeats;
+  return !(number_mutations < mut_rate * repeats * number_of_weights - error ||
+           number_mutations > mut_rate * repeats * number_of_weights + error);
```

Comment:
I think this does what you wrote as well:
```suggestion
  return number_mutations > mut_rate * repeats * number_of_weights - error &&
           number_mutations < mut_rate * repeats * number_of_weights + error);
```